pkg/skaffold/runner/build_deploy.go

```diff
@@ -94,6 +94,15 @@ func (r *SkaffoldRunner) BuildAndTest(ctx context.Context, out io.Writer, artifa
 	return bRes, nil
 }
 
+// Test tests a list of already built artifacts.
+func (r *SkaffoldRunner) Test(ctx context.Context, out io.Writer, artifacts []build.Artifact) error {
```

Comment:
this doesn't necessarily need to be addressed in this PR, but this function doesn't really seem like it should live in a file called `build_deploy.go` :) we should probably split these up into `build.go`, `test.go`, and `deploy.go`.

Reply:
Added a tracking pr - https://github.com/GoogleContainerTools/skaffold/issues/5147